### PR TITLE
I3 us 27

### LINKED ISSUE #30: User Story 27, User Profile displays Orders link
As a registered user
When I visit my Profile page
And I have orders placed in the system
Then I see a link on my profile page called "My Orders"
When I click this link my URI path is "/profile/orders"

### PR DESCRIPTION
**Pull request by D-Lessenden**
Refactored the order controller further by moving some logic into the cart poro. 

Create orders link in the user profile as well as a user profile that displays orders.

Create a test for US 29, a profile/orders show page but is currently commented out because the corresponding code has not been fully written yet.

### CODE REVIEW

File: app/models/cart.rb [hunk 1]
```diff
@@ -35,4 +35,14 @@ def total
   def item_available?(item)
     @contents[item] < Item.find(item).inventory
   end
+
+  def item_orders_create(cart_params, order)
```
Comment (dhalverson) [inline app/models/cart.rb:39]: Model test required here?

**Review by dhalverson — CHANGES_REQUESTED** [review]
Not sure how we would test this?

File: app/models/cart.rb (hunk 1 above)
Comment (D-Lessenden) [reply app/models/cart.rb:39]: Good point. Got it!

File: spec/models/cart_spec.rb [hunk 2]
```diff
@@ -55,5 +66,10 @@
       expect(@cart.item_available?(@hippo.id.to_s)).to eq(false)
       expect(@cart.item_available?(@ogre.id.to_s)).to eq(true)
     end
+
+    it "creates a join table between cart and order" do
```
Comment (dhalverson) [inline spec/models/cart_spec.rb:59]: Interesting test here, thank you for adding it!

**Review by dhalverson — APPROVED** [review]
Thank you for adding the test

File: spec/features/orders/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    it "I can click See Order and go to that order show page" do
+      expect(page).to have_link("Confirmation number: #{@user_1.orders.first.id}")
+      click_on "Confirmation number: #{@user_1.orders.first.id}"
+      expect(current_path).to eq("/profile/orders/#{@user_1.orders.first.id}")
+    end
+
+    it "has the following information" do
+      click_on "Confirmation number: #{@user_1.orders.first.id}"
+      # expect(page).to have_content("#{@user_1.orders.first.id}")
```
Comment (ryandbarnett) [inline spec/features/orders/show_spec.rb:38]: I don't think we should merge commented out code into dev unless we have a good reason.

**Review by ryandbarnett — APPROVED** [review]

**Pull request merged**